recipes-bsp/secure-boot-otp-provisioning/README.md

```diff
@@ -0,0 +1,149 @@
+# IOT2050 secure boot customer key provisioning
+
+IOT2050 Advanced PG2 enables the SoC extend OTP area for storing the customer
+key used for signing the firmware & software that validated by SEBoot and afterwards.
+
+Below are the steps to programming keys into this OTP area.
+
+## Create the keys
+
+First, you need to generate at least two sets of key pair under the folder
+`files/keys`. At most you can have three key pair sets. An example of key pairs
+are already there for reference.
+
+These keys are named `MPK`, `SMPK` and `BMPK` respectively.
+
+```shell
+openssl req -x509 -newkey rsa:4096 -keyout custMpk.pem -nodes -outform pem -out custMpk.crt -sha256
+openssl req -x509 -newkey rsa:4096 -keyout custSmpk.pem -nodes -outform pem -out custSmpk.crt -sha256
+# optional the third key
+#openssl req -x509 -newkey rsa:4096 -keyout custBmpk.pem -nodes -outform pem -out custBmpk.crt -sha256
+```
+
+> :warning:
+> Do remove the example dummy keys under the keys folder!
```

Comment:
With even more keys, this is really not a good pattern anymore. We should rather permit injecting local keys and switch them all at-once or fail.

Reply:
Are you suggesting that we should not provide the `freedom` of `only programming the key without enabling secure boot` to user?